utils/gui.lua

```diff
@@ -18,6 +18,8 @@ local Gui = {}
 local data = {}
 local element_map = {}
 
+Gui.tag = '__@level-RedMew__event_handler_tag__'
```

Comment:
Happy for you to ignore this as you are doing all the work 😅.
But I'd still do something like this as I think it's makes it easier to read/understand the code where the tags is used.
```suggestion
Gui.event_tag = '__@level-RedMew__event_handler_tag__'
```
I'm not too concerned what the string value is as it's unlikely to clash.